test/javascript/components/magellan.js

```diff
@@ -82,7 +84,9 @@ describe('Magellan', function() {
       var hasError = false;
       try {
 				var target = $html.find('a').eq(-1).attr('href');
-	      plugin.scrollToLoc(target); 
+        $html.one('load', function(){
+          plugin.scrollToLoc(target);
+        });
```

Comment:
Same here. No error triggered inside `scrollToLoc` will be triggered here.

See for example:
```js
// Numbers indicate the order in which each line is executed
...                               /* 1 */
try {                            
  setTimeout(/* 2 */ () => {
    throw new Error('err');       /* 7 */
  }, 0);                          /* 3 */
} catch(err) {                    /* [not executed] would be 4 */
  console.log('catched');         /* [not executed] would be 5 */
}
...                               /* 6 */
```

The code above will trigger an error. It will not be catched.

Reply:
True. Can you show or provide the solution? Currently not 100% clear to me how I can solve these things here as this component feels so tricky =/